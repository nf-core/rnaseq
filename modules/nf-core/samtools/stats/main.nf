process SAMTOOLS_STATS {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/samtools:1.19.2--h50ea8bc_0' :
        'biocontainers/samtools:1.19.2--h50ea8bc_0' }"

    input:
    tuple val(meta), path(input), path(input_index)
    tuple val(meta2), path(fasta)

    output:
    tuple val(meta), path("*.stats"), emit: stats
    path  "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def reference = fasta ? "--reference ${fasta}" : ""
    """
    samtools \\
        stats \\
        --threads ${task.cpus} \\
        ${reference} \\
        ${input} \\
        > ${prefix}.stats

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.stats

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
