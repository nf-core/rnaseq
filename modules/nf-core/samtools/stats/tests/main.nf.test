nextflow_process {

    name "Test Process SAMTOOLS_STATS"
    script "../main.nf"
    process "SAMTOOLS_STATS"
    tag "modules"
    tag "modules_nfcore"
    tag "samtools"
    tag "samtools/stats"

    test("bam") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                ])
                input[1] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                {assert process.success},
                {assert snapshot(process.out).match()}
            )
        }
    }

    test("cram") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram', checkIfExists: true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/cram/test.paired_end.recalibrated.sorted.cram.crai', checkIfExists: true)
                ])
                input[1] = Channel.of([
                    [ id:'genome' ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                {assert process.success},
                {assert snapshot(process.out).match()}
            )
        }
    }
}
