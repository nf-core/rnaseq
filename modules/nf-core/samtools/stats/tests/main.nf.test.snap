{
    "cram": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,01812900aa4027532906c5d431114233"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0514ceb1769b2a88843e08c1f82624a9"
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,01812900aa4027532906c5d431114233"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0514ceb1769b2a88843e08c1f82624a9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-02-13T16:15:25.562429714"
    },
    "bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,5d8681bf541199898c042bf400391d59"
                    ]
                ],
                "1": [
                    "versions.yml:md5,0514ceb1769b2a88843e08c1f82624a9"
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.stats:md5,5d8681bf541199898c042bf400391d59"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0514ceb1769b2a88843e08c1f82624a9"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.01.0"
        },
        "timestamp": "2024-02-13T16:15:07.857611509"
    }
}